plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -1597,6 +1608,9 @@ public void write(int b)
         public void write(byte[] bytes, int offset, int length)
                 throws IOException
         {
+            if (!initialized) {
+                initializeBuffer();
+            }
             while (length > 0) {
                 int copied = min(buffer.length - bufferSize, length);
```

Comment:
With the current code, the `buffer` capacity is never increased and stays at `INITIAL_BUFFER_SIZE`.

Reply:
That is not true, it would be increased to `targetPartSize` after reaching the last `else if` statement in `flushBuffer` method. The one you want me to remove in the below comment. But I didn't notice the incorrect condition you mentioned above, my intention was to make it: `if (bufferSize == targetPartSize || (finished && bufferSize > 0)) {` 
Anyway the implementation proposed by you is smoother